src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -1150,6 +1150,35 @@ MsalIdTokenCacheItem ITokenCacheInternal.GetIdTokenCacheItem(MsalAccessTokenCach
             return tenantProfiles;
         }
 
+        async Task<Account> ITokenCacheInternal.GetAccountAssociatedWithAccessTokenAsync(
+            AuthenticationRequestParameters requestParameters,
+            MsalAccessTokenCacheItem msalAccessTokenCacheItem)
+        {
+            if (requestParameters.AuthorityInfo.AuthorityType != AuthorityType.Aad)
+            {
+                return null;
+            }
+
+            Debug.Assert(msalAccessTokenCacheItem.HomeAccountId != null);
```

Comment:
This will work only in the debug mode. You may want to use exception here for all modes.